opentasks-provider/src/main/java/org/dmfs/provider/tasks/model/adapters/DateTimeIterableFieldAdapter.java

```diff
@@ -0,0 +1,207 @@
+        {
+            // if the time stamp list is null we return an empty Iterable
+            return EmptyIterable.instance();
+        }
+
+        String datetimeList = cursor.getString(tdLIdx);
+
+        // create a new TimeZone for the given time zone string
+        String timezoneString = mTimeZoneFieldName == null ? null : cursor.getString(tzIdx);
+        TimeZone timeZone = timezoneString == null ? null : TimeZone.getTimeZone(timezoneString);
+
+        return new DateTimeList(timeZone, datetimeList);
+    }
+
+
+    @Override
+    public Iterable<DateTime> getFrom(Cursor cursor, ContentValues values)
+    {
+        int tsIdx;
+        int tzIdx;
+        String datetimeList;
+        String timeZoneId = null;
+
+        if (values != null && values.containsKey(mDateTimeListFieldName))
+        {
+            if (values.getAsString(mDateTimeListFieldName) == null)
+            {
+                // the date times are null, so we return null
+                return EmptyIterable.instance();
+            }
+            datetimeList = values.getAsString(mDateTimeListFieldName);
+        }
+        else if (cursor != null && (tsIdx = cursor.getColumnIndex(mDateTimeListFieldName)) >= 0)
+        {
+            if (cursor.isNull(tsIdx))
+            {
+                // the date times are null, so we return an empty Iterable.
+                return EmptyIterable.instance();
+            }
+            datetimeList = cursor.getString(tsIdx);
+        }
+        else
+        {
+            throw new IllegalArgumentException("Missing date time list column.");
+        }
+
+        if (mTimeZoneFieldName != null)
+        {
+            if (values != null && values.containsKey(mTimeZoneFieldName))
+            {
+                timeZoneId = values.getAsString(mTimeZoneFieldName);
+            }
+            else if (cursor != null && (tzIdx = cursor.getColumnIndex(mTimeZoneFieldName)) >= 0)
+            {
+                timeZoneId = cursor.getString(tzIdx);
+            }
+            else
+            {
+                throw new IllegalArgumentException("Missing timezone column.");
+            }
+        }
+
+        // create a new TimeZone for the given time zone string
+        TimeZone timeZone = timeZoneId == null ? null : TimeZone.getTimeZone(timeZoneId);
+
+        return new DateTimeList(timeZone, datetimeList);
+    }
+
+
+    @Override
+    public void setIn(ContentValues values, Iterable<DateTime> value)
+    {
+        if (value != null)
+        {
+            String stringValue = new Reduced<>(
+                    new StringBuilder(256),
+                    StringBuilder::append,
+                    new Alternating<>(
+                            new Mapped<>(
+                                    DateTime::toString,
+                                    new Mapped<>(dt -> dt.isFloating() ? dt : dt.shiftTimeZone(DateTime.UTC), value)),
+                            new ConstantGenerator<>(","))).value().toString();
```

Comment:
I would love if we could switch to Kotlin in the near future.. This part would become:
```kotlin
fun setIn(values : ContentValues, dateTimes: Iterable<DateTime>) {

    val datesString = dateTimes
            .map { if (it.isFloating) it else it.shiftTimeZone(DateTime.UTC) }
            .joinToString(",")
    ...
}
```

Reply:
After committing this I noticed that I can just use `TextUtils.join`. I'll change that.

Reply:
Not it's just 
```
String stringValue = TextUtils.join(",", new Mapped<>(dt -> dt.isFloating() ? dt : dt.shiftTimeZone(DateTime.UTC), value));
```

Reply:
That's good, minus 3 classes :)